crates/forge/tests/integration/trace.rs

```diff
@@ -0,0 +1,667 @@
+                trace: Array<CallEntryPoint>, checker_address: ContractAddress
+            ) {
+                let expected_trace = array![
+                    CallEntryPoint {
+                        entry_point_type: EntryPointType::External,
+                        entry_point_selector: selector!("panic"),
+                        calldata: array![],
+                        storage_address: test_address(),
+                        caller_address: 0.try_into().unwrap(),
+                        call_type: CallType::Delegate,
+                    },
+                ];
+            
+                assert(trace == expected_trace, 'checker libcall panic');
+            }
+        "#
+        ),
+        Contract::from_code_path(
+            "TraceInfoProxy".to_string(),
+            Path::new("tests/data/contracts/trace_info_proxy.cairo"),
+        )
+        .unwrap(),
+        Contract::from_code_path(
+            "TraceInfoChecker".to_string(),
+            Path::new("tests/data/contracts/trace_info_checker.cairo"),
+        )
+        .unwrap()
+    );
+
+    let result = run_test_case(&test);
+
+    assert_passed!(result);
+}
+
+#[test]
+fn trace_l1_handler() {
+    let test = test_case!(
+        indoc!(
+            r#"
+            use snforge_std::{declare, ContractClassTrait, test_address, L1HandlerTrait,};
+            use snforge_std::trace::{CallEntryPoint, CallType, EntryPointType, get_last_call_trace};
+            
+            use starknet::ContractAddress;
+            
+            #[test]
+            fn test_l1_handler_call_trace_info() {
+                let proxy = declare('TraceInfoProxy');
+                let checker = declare('TraceInfoChecker');
+            
+                let checker_address = checker.deploy(@array![]).unwrap();
+                let proxy_address = proxy.deploy(@array![checker_address.into()]).unwrap();
+            
+                let mut l1_handler = L1HandlerTrait::new(checker_address, function_name: 'handle_l1_message');
+            
+                l1_handler.from_address = 123;
+                l1_handler.payload = array![proxy_address.into()].span();
+            
+                l1_handler.execute().unwrap();
+                assert_trace_after_l1_handler_call(get_last_call_trace(), proxy_address, checker_address);
+            }
+            
+            fn assert_trace_after_l1_handler_call(
+                trace: Array<CallEntryPoint>, proxy_address: ContractAddress, checker_address: ContractAddress
+            ) {
+                let expected_trace = array![
+                    CallEntryPoint {
+                        entry_point_type: EntryPointType::L1Handler,
+                        entry_point_selector: selector!("handle_l1_message"),
+                        calldata: array![123, proxy_address.into()],
+                        storage_address: checker_address,
+                        caller_address: test_address(),
```

Comment:
Not sure if this one behaves as intended - what should the caller address really be during L1 call?

Reply:
It should be 0x0, will fix that
https://docs.starknet.io/documentation/architecture_and_concepts/Network_Architecture/starknet-state/#address_0x0